Lib/test/test_largefile.py

```diff
@@ -151,9 +151,22 @@ def test_seekable(self):
                 self.assertTrue(f.seekable())
 
 
+def skip_no_disk_space(path, required):
+    def decorator(fun):
+        def wrapper(*args, **kwargs):
+            if shutil.disk_usage(path).free < required:
+                hsize = int(required / 1024 / 1024)
+                raise unittest.SkipTest(
+                    f"required {hsize} MiB of free disk space")
+            return fun(*args, **kwargs)
+        return wrapper
+    return decorator
+
+
 class TestCopyfile(LargeFileTest, unittest.TestCase):
     open = staticmethod(io.open)
 
+    @skip_no_disk_space(os.getcwd(), size * 3)
```

Comment:
os.getcwd() can be different when the function is defined and when the test is executed. I suggested to call skip_no_disk_space() inside the function. Something like:

```
def test_it(self):
    skip_no_disk_space(TESTFN, size * 3)
    ...
```

TESTFN is usually relative. I'm not sure if skip_no_disk_space() should be made absolute  using os.path.abspath().

Reply:
Good point. I used os.path.realpath instead.